src/Identity/Extensions.Core/src/IdentityBuilder.cs

```diff
@@ -71,15 +71,39 @@ private IdentityBuilder AddScoped(Type serviceType, Type concreteType)
         /// <typeparam name="TValidator">The user validator type.</typeparam>
         /// <returns>The current <see cref="IdentityBuilder"/> instance.</returns>
         public virtual IdentityBuilder AddUserValidator<TValidator>() where TValidator : class
-            => AddScoped(typeof(IUserValidator<>).MakeGenericType(UserType), typeof(TValidator));
+        {
```

Comment:
Can you introduce a helper method that takes identityType, customType so that all of these methods can share the same code

Reply:
I can do that, however for the InvalidOperationException, it currently uses a clean "IInterface" string.  Whereas if I pull this into a helper, I need to either do string parsing on the generic type: ```typeName.Substring(0, typeName.IndexOf("`")``` to lop off the generic tilda garbage at the end or pass in a string value as an additional parameter.  Which would you prefer?

Reply:
I would just pass in an additional string value to make it easier, thanks!